src/backend/aspen/workflows/import_pango_lineages/import.py

```diff
@@ -0,0 +1,198 @@
+        (note, that's a tab character after `Q.3`)
+    Some lineages are "Withdrawn". This is denoted by a `*` at lineage start
+        *B.1.1.156	Withdrawn: South African lineage
+        (note, that's a tab character after `*B.1.1.156`)
+
+    In general, from talking with Alli Black, it appears that we should be fine
+    to only ever care about non-withdrawn lineages. Similarly, we can ignore
+    any "alias" names mentioned (eg, `B.1.1.7.3` above), the alias name is more
+    of an academic understanding, not something used for identification.
+
+    Returns:
+        str OR None: Lineage for the line OR None if line should be ignored
+    """
+    lineage_chunk = line.split()[0]
+    if exclude_withdrawn and lineage_chunk[0] == "*":
+        return None
+    return lineage_chunk
+
+
+def safety_check_first_line(first_line: str) -> None:
+    """Inspects first line of lineage_notes.txt to perform safety check.
+
+    We pull all of our Pango lineages from a human-edited .txt file.
+    The format has been stable so far, but if things ever change significantly,
+    the planned loading process will probably explode. In case of changes,
+    we will need to investigate and alter the loading process.
+
+    This check exists to avoid ever accidentally loading a file where the
+    format has (probably) changed. Assumption is that if the first line has
+    changed from what it used to be, the file format has probably changed.
+    Will print the problem and raise an exception.
+
+    Raises:
+        RuntimeError -- If first line of file not what was expected.
+    """
+    EXPECTED_FIRST_LINE = "Lineage\tDescription\n"
+    if first_line != EXPECTED_FIRST_LINE:
+        print("First line of imported lineages file has changed!")
+        print("Loading script was originally written for previous version.")
+        print(f"Expected: '{EXPECTED_FIRST_LINE}'")
+        print(f"Actually got first line: '{first_line}'")
+        print("Very likely you need to rewrite loading script. Aborting.")
+        raise RuntimeError("Format of lineage file has likely changed")
+    return None
+
+
+def get_lineages(lineage_notes_file: io.TextIOBase) -> list[str]:
+    """Extracts all current Pango lineages from lineage_notes file.
+
+    Talking to Pangolin folks, they said the best canonical listing of all
+    current lineages is to use their (manually updated) lineage_notes.txt
+    https://github.com/cov-lineages/pango-designation/issues/456
+
+    This function takes that file and parses out list of lineages.
+    """
+    # First line is header, no lineage data, but we use as a safety check.
+    first_line = next(lineage_notes_file)
+    safety_check_first_line(first_line)
+    # Now we parse remainder of lines from file and extract lineages
+    return [
+        lineage
+        for line in lineage_notes_file
+        if (lineage := extract_lineage_from_line(line)) is not None
```

Comment:
This is the first time I have ever used a walrus operator. I'll be honest, I was only vaguely aware of their existence before, but I learned about it because I was getting a mypy error from my original version I wrote that was like this:
```
[
extract_lineage_from_line(line)
for line in lineage_notes_file
if extract_lineage_from_line(line)
]
```
Basically the guarantees from the `if` conditional were clear to me the dev, but not really usable by mypy. But I actually like the walrus operator way of writing it! It feels remarkably readable to me while also solving the mypy problem I was having. (also yeah, I totally could have done this with `[].append` stuff, but list comprehensions are so pythonnnnnniiiiiccc)

If you were like me and had not previously interacted with them before, here's a good writeup of how I'm using it
https://realpython.com/python-walrus-operator/#list-comprehensions

Reply:
Thanks for the writeup! I'm also only vaguely aware of weird python operators.